api/src/test/java/com/epam/pipeline/manager/pipeline/FolderTemplateManagerTest.java

```diff
@@ -75,70 +62,27 @@
     private static final String CHILD_TEMPLATE_FOLDER_NAME_1 = "test-1";
     private static final String DATASTORAGE_NAME_1 = "ds-1";
 
-    @Autowired
+    @SpyBean
     private FolderTemplateManager folderTemplateManager;
     @Autowired
     private FolderManager folderManager;
     @Autowired
     private MetadataManager metadataManager;
-    @Autowired
+    @MockBean
     private DataStorageManager dataStorageManager;
-    @SpyBean
-    private S3StorageProvider storageProviderManager;
-    @Autowired
-    private AclTestDao aclTestDao;
-    @Autowired
+    @MockBean
     private GrantPermissionManager permissionManager;
```

Comment:
Spring test context is reused between several test classes using the same configuration. In this case all test classes inheriting from `AbstractAclTest` are using just the same spring context and therefore are sharing same beans.

If you look to `com.epam.pipeline.test.acl.AclTestBeans` class you would see that `GrantPermissionManager` is already defined there as a spy. Therefore you can't mock this bean in `FolderTemplateManagerTest`.

What you can do is you can mock the beans that `GrantPermissionManager` depends on.

Be cautious with using shared test configurations because changes in one test class may affect others.

Reply:
Fixed